odoo/addons/base/ir/ir_model.py

```diff
@@ -287,6 +287,12 @@ def _check_selection(self, selection):
             raise UserError(_("The Selection Options expression is not a valid Pythonic expression."
                               "Please provide an expression in the [('key','Label'), ...] format."))
 
+    @api.constrains('domain')
+    def _check_domain(self):
+        for field in self:
+            if field.domain:
+                safe_eval(field.domain or '[]')
```

Comment:
@Elkasitu 
Why do you need `field.domain or '[]'` as you are already checking `if field.domain`
(or You meant `safe_eval(field.domain) or '[]'` )